### PR TITLE
feat: Implement validation for eligibility flag

### PR DESCRIPTION
**Pull request by sudipnhs**
## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [x] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: tests/ScreeningValidationServiceTests/StaticValidation/StaticValidationTests.cs [hunk 1]
```diff
@@ -1035,4 +1035,48 @@ public async Task Run_CurrentPostingAndPrimaryCareProvider_DoesNotCreateExceptio
...
+        _handleException.Verify(handleException => handleException.CreateValidationExceptionLog(
+            It.Is<IEnumerable<RuleResultTree>>(r => r.Any(x => x.Rule.RuleName == "94.EligibilityFlag.NonFatal")),
+            It.IsAny<ParticipantCsvRecord>()),
+            Times.Once());
+    }
+
+    [TestMethod]
+    [DataRow(Actions.New, "1")]
+    [DataRow(Actions.Removed, "0")]
+    public async Task Run_Should_Not_Create_Exception_When_EligibilityFlag_Is_Invalid_For_RecordType(string recordtype, string eligibilityflag)
```
Comment (Warren-Pitterson) [inline tests/ScreeningValidationServiceTests/StaticValidation/StaticValidationTests.cs:1064]: Same with this method name please

Comment (sudipnhs) [reply tests/ScreeningValidationServiceTests/StaticValidation/StaticValidationTests.cs]: Thanks for the feedback. I have now rectified it.

File: tests/ScreeningValidationServiceTests/StaticValidation/StaticValidationTests.cs [hunk 2]
```diff
@@ -1035,4 +1035,48 @@ public async Task Run_CurrentPostingAndPrimaryCareProvider_DoesNotCreateExceptio
             Times.Never());
     }
     #endregion
+
+    #region Validate Eligibility Flag as per Record Type (Rule 94)
+    [TestMethod]
+    [DataRow(Actions.New, "0")]
+    [DataRow(Actions.Removed, "1")]
+    public async Task Run_Should_Create_Exception_When_EligibilityFlag_Is_Invalid_For_RecordType(string recordtype, string eligibilityflag)
```
[the remark this answers is not in the document]
Comment (sudipnhs) [reply tests/ScreeningValidationServiceTests/StaticValidation/StaticValidationTests.cs]: Thanks for the feedback. I have now rectified it.

**Review by Warren-Pitterson — DISMISSED** [review]

File: tests/ScreeningValidationServiceTests/StaticValidation/StaticValidationTests.cs [hunk 3]
```diff
@@ -1035,4 +1035,48 @@ public async Task Run_CurrentPostingAndPrimaryCareProvider_DoesNotCreateExceptio
...
+        // Assert
+        _handleException.Verify(handleException => handleException.CreateValidationExceptionLog(
+            It.Is<IEnumerable<RuleResultTree>>(r => r.Any(x => x.Rule.RuleName == "94.EligibilityFlag.NonFatal")),
+            It.IsAny<ParticipantCsvRecord>()),
+            Times.Once());
+    }
+
+    [TestMethod]
+    [DataRow(Actions.New, "1")]
+    [DataRow(Actions.Removed, "0")]
```
Comment (andrewmorris43) [inline tests/ScreeningValidationServiceTests/StaticValidation/StaticValidationTests.cs:1063]: Could we include another row here to test the amended record works as expected too?

Comment (sudipnhs) [reply tests/ScreeningValidationServiceTests/StaticValidation/StaticValidationTests.cs:1063]: Datarow added for Amended. 
Resolved

**Pull request closed without merging**